error.go

```diff
@@ -33,6 +36,98 @@ func (e Error) GetID() string {
 	return e.ID
 }
 
+type ErrorLinks struct {
+	About string `json:"about,omitempty"`
+}
+
+type ErrorSource struct {
+	Pointer   string `json:"pointer,omitempty"`
+	Parameter string `json:"parameter,omitempty"`
+}
+
+func (e Error) MarshalJSON() ([]byte, error) {
+	// Collect the fields we need to marshal, all are optional.
+	var fields []string
+
+	if e.ID != "" {
+		fields = append(fields, fmt.Sprintf(`"id":%q`, e.ID))
+	}
+
+	if e.Links.About != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"links":{"about":%q}`, e.Links.About))
+	}
+
+	if e.Status != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"status":%q`, e.Status))
+	}
+
+	if e.Code != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"code":%q`, e.Code))
+	}
+
+	if e.Title != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"title":%q`, e.Title))
+	}
+
+	if e.Detail != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"detail":%q`, e.Detail))
+	}
+
+	if e.Source.Pointer != "" && e.Source.Parameter != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"source":{"pointer":%q,"parameter":%q}`,
+				e.Source.Pointer, e.Source.Parameter))
+	} else if e.Source.Pointer != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"source":{"pointer":%q}`, e.Source.Pointer))
+	} else if e.Source.Parameter != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"source":{"parameter":%q}`, e.Source.Parameter))
+	}
+
+	// Capture our metadata using json.Marshal on the interface{}
+	if e.Meta != nil {
+		b, err := json.Marshal(e.Meta)
+		if err != nil {
+			return nil, err
+		}
+
+		fields = append(fields, `"meta":`+string(b))
+	}
+
+	// Generate our JSON
+	var buf bytes.Buffer
+	if _, err := buf.WriteString("{"); err != nil {
+		return nil, err
+	}
+
+	first := true
+	for _, field := range fields {
```

Comment:
you were a little bit faster ;) I misunderstood the requirement that omitempty "didn't work". 

But at least that way there is no "manual" encoding. That is always the wrong thing to do :D